app/src/main/cpp/HandMeshRenderer.h

```diff
@@ -0,0 +1,61 @@
+/* -*- Mode: C++; tab-width: 20; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#pragma once
+
+#include "vrb/CreationContext.h"
+
+namespace crow {
+
+enum class HandMeshStrategy { None, Skinned, Spheres };
+
+struct Controller;
+
+class HandMeshRenderer;
+typedef std::shared_ptr<HandMeshRenderer> HandMeshRendererPtr;
+
+class HandMeshRenderer {
+protected:
+    vrb::CreationContextWeak context;
+    bool enabled = false;
+    virtual void Shutdown() { };
+public:
+    virtual void Initialize(vrb::CreationContextPtr& aContext) { context = aContext; };
```

Comment:
As far as I see only subclasses call this method so we don't need it to be virtual. I think you should pass the context to the create() and construct methods and let subclasses do this assignment in the constructor

Reply:
Ok I can try that. As said earlier I think there is value in having the Initialize()/Shutdown() pattern, but I agree we could save some code.

Reply:
No need for it to be virtual nor protected

Reply:
At the end I removed the Initialize(), and moved its contents to the respective constructors. If we don't have Shutdown(), the argument for having Initialize() weakens.